resources/fetching/fetchers/gcp/assets_fetcher.go

```diff
@@ -144,6 +144,14 @@ func (r *GcpAsset) GetMetadata() (fetching.ResourceMetadata, error) {
 
 func (r *GcpAsset) GetElasticCommonData() any { return nil }
 
+// a GCP asset name is made up of its ancestors
+// the resource id is the last part of the name
+// see https://cloud.google.com/apis/design/resource_names#resource_id
```

Comment:
One problem I can see is that it might create duplicate resource IDs, leading to clashes in IDs. This could result in missing findings from the latest findings index.

Reply:
ancestors?

Reply:
yaani parents?

Reply:
Do you mean we should represent the direct parent in the resource id?